src/migration/1723729297617-migration.ts

```diff
@@ -0,0 +1,171 @@
+                    CREATE TABLE import (
+                        id UUID DEFAULT gen_random_uuid() PRIMARY KEY,
+                        revision_id UUID,
+                        mime_type VARCHAR(255),
+                        filename VARCHAR(255),
+                        hash VARCHAR(255),
+                        uploaded_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+                        type import_type NOT NULL,
+                        location location_type NOT NULL,
+                        FOREIGN KEY (revision_id) REFERENCES revision(id) ON DELETE CASCADE
+                    );
+
+                    CREATE TABLE csv_info (
+                        import_id UUID PRIMARY KEY,
+                        delimiter CHAR(1),
+                        quote CHAR(1),
+                        linebreak VARCHAR(2),
+                        FOREIGN KEY (import_id) REFERENCES import(id) ON DELETE CASCADE
+                    );
+
+                    CREATE TYPE source_action_type AS ENUM ('create', 'append', 'truncate-then-load', 'ignore');
+
+                    CREATE TABLE source (
+                        id UUID DEFAULT gen_random_uuid() PRIMARY KEY,
+                        dimension_id UUID,
+                        import_id UUID UNIQUE,
+                        revision_id UUID,
+                        column_index INT,
+                        csv_field TEXT,
+                        action source_action_type NOT NULL,
+                        FOREIGN KEY (dimension_id) REFERENCES dimension(id) ON DELETE CASCADE,
+                        FOREIGN KEY (import_id) REFERENCES import(id) ON DELETE CASCADE,
+                        FOREIGN KEY (revision_id) REFERENCES revision(id) ON DELETE CASCADE
+                    );
+
+                    INSERT INTO users (
+                        id,
+                        email,
+                        oidc_subject,
+                        oidc_issuer,
+                        access_token,
+                        refresh_token,
+                        id_token,
+                        token_expiry,
+                        name,
+                        given_name,
+                        last_name,
+                        profile_picture,
+                        created_at,
+                        updated_at,
+                        active
+                    )
+                    VALUES (
+                        '12345678-1234-1234-1234-123456789012',
+                        'test@test.com',
+                        '',
+                        'localAuth',
+                        '',
+                        '',
+                        '',
+                        NULL,
+                        'Test User',
+                        'Test',
+                        'User',  -- Corrected by closing the quote
+                        '',
+                        CURRENT_TIMESTAMP,
+                        CURRENT_TIMESTAMP,
+                        true
+                    );
```

Comment:
I would prefer not to add test data in the migrations as we will probably want to run these in prod eventually. We can load test data from the tests before they run, or just manually if it's for manual testing.

Reply:
Again chicken and egg problem...We need a test user we can put things against while we workout the rest of the user journey.  I am happy to remove this once that work is in to place and we ensure we have a way to provisions some initial users.